pkg/operator/controller/status/controller_test.go

```diff
@@ -383,63 +391,63 @@ func TestComputeOperatorStatusVersions(t *testing.T) {
 		expectedVersions      versions
 	}{
 		{
-			description:           "initialize versions, operator is avaiable",
-			oldVersions:           versions{UnknownVersionValue, UnknownVersionValue},
-			curVersions:           versions{"v1", "ic-v1"},
+			description:           "initialize versions, operator is available",
+			oldVersions:           versions{UnknownVersionValue, UnknownVersionValue, UnknownVersionValue},
+			curVersions:           versions{"v1", "ic-v1", "c-v1"},
 			allIngressesAvailable: true,
-			expectedVersions:      versions{"v1", "ic-v1"},
+			expectedVersions:      versions{"v1", "ic-v1", "c-v1"},
 		},
 		{
 			description:      "initialize versions, operator is not available",
-			oldVersions:      versions{UnknownVersionValue, UnknownVersionValue},
-			curVersions:      versions{"v1", "ic-v1"},
-			expectedVersions: versions{UnknownVersionValue, UnknownVersionValue},
+			oldVersions:      versions{UnknownVersionValue, UnknownVersionValue, UnknownVersionValue},
+			curVersions:      versions{"v1", "ic-v1", "c-v1"},
+			expectedVersions: versions{UnknownVersionValue, UnknownVersionValue, UnknownVersionValue},
 		},
 		{
 			description:           "update with no change",
-			oldVersions:           versions{"v1", "ic-v1"},
-			curVersions:           versions{"v1", "ic-v1"},
+			oldVersions:           versions{"v1", "ic-v1", "c-v1"},
+			curVersions:           versions{"v1", "ic-v1", "c-v1"},
 			allIngressesAvailable: true,
-			expectedVersions:      versions{"v1", "ic-v1"},
+			expectedVersions:      versions{"v1", "ic-v1", "c-v1"},
 		},
 		{
 			description:      "update operator version, operator is not available",
-			oldVersions:      versions{"v1", "ic-v1"},
-			curVersions:      versions{"v2", "ic-v1"},
-			expectedVersions: versions{"v1", "ic-v1"},
+			oldVersions:      versions{"v1", "ic-v1", "c-v1"},
+			curVersions:      versions{"v2", "ic-v1", "c-v2"},
```

Comment:
No harm in having this test case for the status logic, but in the broader perspective, this should never happen, right?  The "current" ingresscontroller and canary images should always be equal.